File: 2551-apply-operations-to-an-array/apply-operations-to-an-array.py
class Solution:
    def applyOperations(self, nums: List[int]) -> List[int]:
        n = len(nums)
        for i in range(n-1):
            if nums[i] == nums[i+1]:
                nums[i]*=2
                nums[i+1] = 0
        output = [0]*n # we fill the array with zero and later override
        index = 0 # to control the index of the output
        for i in range(n):
            if nums[i]!=0:
                output[index] = nums[i]
                index += 1
        return output

File: 2551-apply-operations-to-an-array/apply-operations-to-an-array.ts
function applyOperations(nums: number[]): number[] {
    const arrayLength: number = nums.length;
    for (let i = 0; i < arrayLength - 1; i++) {
        if (nums[i] === nums[i + 1]) {
            nums[i] = nums[i] << 1;
            nums[i + 1] = 0;
        }
    }
  
    const resultArray: number[] = Array(arrayLength).fill(0);
  
    let currentIndex: number = 0;
  
    for (const element of nums) {
        if (element !== 0) {
            resultArray[currentIndex] = element;
            currentIndex++;
        }
    }
  
    return resultArray;
};